components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,547 @@
+    private val fetchScope: CoroutineScope = delegate.fetchScope
+
+    private val errorReporter = delegate.errorReporter
+
+    private val logger = delegate.logger
+
+    private val nimbusClient: NimbusClientInterface
+
+    override var globalUserParticipation: Boolean
+        get() = nimbusClient.getGlobalUserParticipation()
+        set(active) {
+            dbScope.launch {
+                setGlobalUserParticipationOnThisThread(active)
+            }
+        }
+
+    init {
+        // Set the name of the native library so that we use
+        // the appservices megazord for compiled code.
+        System.setProperty(
+            "uniffi.component.nimbus.libraryOverride",
+            System.getProperty("mozilla.appservices.megazord.library", "megazord")
+        )
+        // Build a File object to represent the data directory for Nimbus data
+        val dataDir = File(context.applicationInfo.dataDir, NIMBUS_DATA_DIR)
+
+        // Build Nimbus AppContext object to pass into initialize
+        val experimentContext = buildExperimentContext(context, appInfo, deviceInfo)
+
+        // Initialize Nimbus
+        val remoteSettingsConfig = server?.let {
+            RemoteSettingsConfig(
+                serverUrl = it.url.toString(),
+                bucketName = it.bucket,
+                collectionName = it.collection
+            )
+        }
+
+        nimbusClient = NimbusClient(
+            experimentContext,
+            dataDir.path,
+            remoteSettingsConfig,
+            // The "dummy" field here is required for obscure reasons when generating code on desktop,
+            // so we just automatically set it to a dummy value.
+            AvailableRandomizationUnits(clientId = null, dummy = 0)
+        )
+    }
+
+    // This is currently not available from the main thread.
+    // see https://jira.mozilla.com/browse/SDK-191
+    @WorkerThread
+    override fun getActiveExperiments(): List<EnrolledExperiment> =
+        nimbusClient.getActiveExperiments()
+
+    @WorkerThread
+    override fun getAvailableExperiments(): List<AvailableExperiment> =
+        nimbusClient.getAvailableExperiments()
+
+    override fun getExperimentBranch(experimentId: String): String? {
+        recordExposure(experimentId)
+        return nimbusClient.getExperimentBranch(experimentId)
+    }
+
+    @WorkerThread
+    override fun getExperimentBranches(experimentId: String): List<Branch>? = withCatchAll {
+        nimbusClient.getExperimentBranches(experimentId)
+    }
+
+    // Method and apparatus to catch any uncaught exceptions
+    @SuppressWarnings("TooGenericExceptionCaught")
+    private fun <R> withCatchAll(thunk: () -> R) =
+        try {
+            thunk()
+        } catch (e: Throwable) {
+            try {
+                errorReporter("Error in Nimbus Rust", e)
+            } catch (e1: Throwable) {
+                logger("Exception calling rust: $e")
+                logger("Exception reporting the exception: $e1")
```

Comment:
nit: see other PR, these used to be `errors` now they're just `infos`.

Reply:
Everything that used to report an error via the `logger.error()` is now doing it via an `errorReporter`. 

Except here where an exception is caught coming from `errorReporter`. If `errorReporter` is itself erroring, I'm guessing nimbus won't be the only one finding out about it, and that will be visible from more than just here.